src/common/io.go

```diff
@@ -0,0 +1,16 @@
+package common
+
+import "io"
+
+func WriteAll(writer io.Writer, data []byte) (n int, err error) {
```

Comment:
Call the first returned value written instead of n because you have another variable inside the function called n and it's confusing